### PR TITLE
Upgrade go-netbox to Netbox version 3.0.2

### PR DESCRIPTION
**Pull request by chdxD1**
This is my first PR here but I hope that this is fine!

This PR tries to update this library to Netbox version 3.0.2. There are some changes however from the official swagger.json file that can be fetched from a Netbox:
 * The maximumInt for scopeID in a VLANGroup was changed to an int. `swagger generate client` generates it in a scientific notation which does not work with go
 * I've removed all `"additionalProperties": {"type": "string", "x-nullable": true}` . This will result in a map of `interface{}` instead of the correct nested types. Swagger does not really support discriminators on the same level (like connected_endpoint_type / cable_peer_type) apparently. However the connected_endpoint and cable_peer_type can vary. Also some nested types are missing (e.g. for console ports).

I think the swagger autogeneration on the netbox side itself also needs some love to not require these changes but it's something someone needs to take care off (https://github.com/netbox-community/netbox/issues/4808, https://github.com/netbox-community/netbox/issues/4804)

### CODE REVIEW

**Comment by awlx** [comment]
@chdxD1 thanks for your PR :). Is it good to merge from your perspective or anything you'd like to add before we merge it?

**Comment by chdxD1** [comment]
@awlx is the failing go 1.16.x check something to worry about? I've tried to replicate it locally with a clean 1.16 docker container but there it worked fine.
Maybe some feedback on the wording in the README would be appreciated https://github.com/netbox-community/go-netbox/pull/121/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5 :)

**Comment by awlx** [comment]
I don't think it's something to worry about as it fails for everything ... and I didn't figure out yet why this happens. Also it's only 1.16.

File: README.md [hunk 1]
```diff
@@ -3,10 +3,10 @@ go-netbox
 
 [![GoDoc](http://godoc.org/github.com/netbox-community/go-netbox?status.svg)](http://godoc.org/github.com/netbox-community/go-netbox) [![Build Status](https://github.com/netbox-community/go-netbox/workflows/main/badge.svg?branch=master)](https://github.com/netbox-community/go-netbox/actions) [![Report Card](https://goreportcard.com/badge/github.com/netbox-community/go-netbox)](https://goreportcard.com/report/github.com/netbox-community/go-netbox)
 
-Package `netbox` provides an API 2.0 client for [netbox-community's NetBox](https://github.com/netbox-community/netbox)
+Package `netbox` provides an API 3.0 client for [netbox-community's NetBox](https://github.com/netbox-community/netbox)
 IPAM and DCIM service.
 
-This package assumes you are using NetBox 2.0, as the NetBox 1.0 API no longer exists.
+This package assumes you are using NetBox 3.0, as the NetBox 1.0 API no longer exists. However NetBox 2.0 might still work as the API isn't that different between 2.0 and 3.0.
```
Comment (awlx) [inline README.md:9]:
```suggestion
This package assumes you are using NetBox 3.0, as the NetBox 1.0 API no longer exists. It wasn't tested on NetBox 2.0 but given the minor changes to the API between 2.0 and 3.0 it still might work.
```

**Review by awlx — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (awlx) [reply README.md:9]: Is that better? Dunno :D.

Comment (chdxD1) [reply README.md]: Sounds good (and "not tested" is the reality anyways). I've also added a short reference to the 2.11 tag.

**Comment by awlx** [comment]
https://github.com/netbox-community/go-netbox/releases/tag/netbox_v3.0 released! Thanks again @chdxD1

**Pull request merged**